benchmarks/distributed/rpc/parameter_server/servers/ParameterServerBase.py

```diff
@@ -0,0 +1,101 @@
+import functools
+import time
+from abc import ABC, abstractmethod
+
+import torch
+from metrics.MetricsLogger import MetricsLogger
+
+
+class ParameterServerBase(ABC):
+
+    PARAMETER_SERVER_BATCH_METRIC = "parameter_server_batch_metric"
+    PARAMETER_SERVER_STRAGGLER_METRIC = "parameter_server_straggler_metric"
+    BP_LOC_STRAGGLER = "bp_location_staggler"
+    BP_LOC_BATCH = "bp_location_batch"
+
+    def __init__(self, rank):
+        self.__metrics_logger = MetricsLogger(rank)
+
+    @abstractmethod
+    def process_gradient(self):
+        return
+
+    @staticmethod
+    @abstractmethod
+    def reset_state():
+        return
+
+    @staticmethod
+    @abstractmethod
+    def get_metrics_rpc():
+        return
+
+    def record_start(self, type, key, name, cuda=True):
+        self.__metrics_logger.record_start(
+            type,
+            key,
+            name,
+            cuda
+        )
+
+    def record_end(self, type, key):
+        self.__metrics_logger.record_end(
+            type,
+            key
+        )
+
+    def record_straggler_start(self, key, cuda=True):
```

Comment:
could you please add some docs to these method to explain what does straggler and batch mean? And what are their start and end time instances?

Reply:
okay.